include/aws_fota.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ *SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**@file
+ *@brief AWS FOTA library header.
+ */
+
+#ifndef AWS_FOTA_H__
+#define AWS_FOTA_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/* Enum to keep the fota status */
+enum fota_status {
+	NONE = 0
+	DOWNLOAD_FIRMWARE,
+	APPLY_FIRMWARE,
+};
+
+/* Map of fota status to report back */
+const char *fota_status_map[] = {
+	[DOWNLOAD_FIRMWARE] = "download_firmware",
+	[APPLY_FIRMWARE] = "apply_update",
+	[NONE] = "none"
+};
+
+enum aws_fota_evt_id {
+	/** AWS FOTA complete and status reported to job document */
+	AWS_FOTA_EVT_DONE,
+	/** AWS FOTA error */
+	AWS_FOTA_EVT_ERROR
+
+}
+
+typedef void (*aws_fota_callback_t)(enum aws_fota_evt_id evt_id);
+
```

Comment:
No API at all?

Reply:
Haven't added it into the header yet.
The goal is for the API to be two functions
`aws_jobs_fota_init(app_version_string, callback)` and `aws_jobs_fota_mqtt_evt_handler(mqtt_client, mqtt_evt)` so the `aws_jobs_fota_mqtt_evt_handler` will have to be called from the applications `mqtt_evt_handler` and just forward the event to the library, then after the library returns handle the `mqtt_evt`.

Another point is to give a value for the application to discard it the current evt. So a goal should be to return an value for discarding the message in application up on receiving relevant `mqtt_messages` on relevant `topics` for the library. This will of course not be possible for `CONNECT`, `DISCONNECT` evt etc.